app/cards/renaissance/flag_bearer.js

```diff
@@ -0,0 +1,62 @@
+FlagBearer = class FlagBearer extends Card {
+
+    types() {
+        return ['action']
+    }
+
+    coin_cost() {
+        return 4
+    }
+
+    artifact_name() {
+        return 'Flag'
+    }
+
+    get_artifact_instance() {
+        return new Flag()
+    }
+
+    play(game, player_cards) {
+        let gained_coins = CoinGainer.gain(game, player_cards, 2)
+        game.log.push(`&nbsp;&nbsp;<strong>${player_cards.username}</strong> gets +$${gained_coins}`)
+    }
+
+    get_artifact(game, player_cards) {
+        let self = this;
+
+        let artifact_index = _.findIndex(player_cards.artifacts, function (artifact) {
+            return artifact.name === self.artifact_name()
+        })
+
+        if (artifact_index === -1) {
+            let ordered_player_cards = TurnOrderedPlayerCardsQuery.turn_ordered_player_cards(game, player_cards)
+            ordered_player_cards.shift()
+
+            _.each(ordered_player_cards, function (other_player_cards) {
+                let other_player_artifact_index = _.findIndex(other_player_cards.artifacts, function (artifact) {
+                    return artifact.name === self.artifact_name()
+                })
+                if (other_player_artifact_index !== -1) {
+                    other_player_cards.artifacts.splice(other_player_artifact_index, 1)
+                }
+                PlayerCardsModel.update(game._id, other_player_cards)
+            })
+
+            let artifact = (this.get_artifact_instance()).to_h()
```

Comment:
I'm curious about the reasoning behind the `get_artifact_instance` method. I don't really have an issue with it, but as is it doesn't seem like much of a gain to break it out into it's own method.

Is there a plan to refactor the entire `get_artifact` method into the `Card` class? I could see this being useful in that situation potentially.

Reply:
I exposed `get_artifact` as a separate method because this way I could just copy/paste content of the card to another file when creating new cards with artifacts. I thought there will be way more of those in the final release, but at the end there was just one more artifact card (`Border Guard`) so it didn't help a lot.